crates/bevy_animation/src/blend.rs

```diff
@@ -0,0 +1,189 @@
+//! This module contains traits for additive blending, for use in animation.
+
+use bevy_color::{
+    ColorToComponents, Gray, Hsla, Hwba, Laba, Lcha, LinearRgba, Oklaba, Oklcha, Srgba, Xyza,
+};
+use bevy_math::*;
+use bevy_transform::components::Transform;
+
+/// A type with a natural method of addative blending.
+///
+/// Any type with a well-behaved method of composition which also implements the [`Interpolate`] trait
+/// can support addative blending. Specifically, the type must:
+///
+/// 1. Have a natural method of composition. For vectors this is addition, for quaternions or rotations
+///    composition is usually expressed as multiplication. We'll use `comp(a, b)` to denote this operation.
+///
+/// 2. The operation must be associative. That is, `comp(comp(a, b), c)` must be equivalent to
+///    `comp(a, comp(b, c))`. The result does not need to be data-identical, but it should be equivalent
+///    under some reasonable notion of equivalence.
+///
+/// 2. Have an `IDENTITY` value such that composition with the identity is equivalent in some way to the
+///    original value.
+///
+/// 3. The value of `T::blend(a, b, w)` should be equivalent to `comp(a, interp(IDENTITY, b, w))`. This implies
+///    that `T::blend(a, b, 0)` is equivalent to `a` and `T::blend(a, b, 1)` is equivalent to `comp(a, b)`.
+///
+/// Some of you will have noticed that these rules encodes the axioms for a Monoid; In fact this trait
+/// represents something similar to a Lie Group.
+pub trait Blend: Interpolate {
+    /// The default value of the blend, which has no effect when blended with other values.
+    const IDENTITY: Self;
+
+    /// Addatively blends another value on top of this one.
+    fn blend(self, other: Self, blend_weight: f32) -> Self;
```

Comment:
The type docstring talks about the properties the blended object should have, but what properties should this `blend` function itself have and how do they relate?